.github/workflows/code_analysis.yml

```
@@ -4,11 +4,25 @@ on:
   group: ${{ github.workflow }}-${{ github.ref_name }}
   cancel-in-progress: true
+permissions:
+  contents: write
+  actions: write
+  pull-requests: write
 jobs:
+  pre-commit:
+    name: Pre-commit
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)